apps/agent-service/src/agent-service.service.ts

```diff
@@ -160,33 +169,52 @@ export class AgentServiceService {
       let orgData;
       if (!user?.userId && agentSpinupDto?.platformAdminEmail) {
 
+
+        // Get Platform admin user by platform admin email
         platformAdminUser = await this.agentServiceRepository.getPlatfomAdminUser(agentSpinupDto?.platformAdminEmail);
 
         userId = platformAdminUser?.id;
       } else {
-        userId = user?.userId;
+        userId = user?.id;
       }
 
+      // Get platform org
       const platformAdminOrgDetails = await this.agentServiceRepository.getPlatfomOrg(agentSpinupDto?.orgName);
+
       if (agentSpinupDto.orgId) {
+
+        // Get organization details
         getOrgAgent = await this.agentServiceRepository.getAgentDetails(agentSpinupDto.orgId);
+
+        // Get organization data by orgId
         orgData = await this.agentServiceRepository.getOrgDetails(agentSpinupDto.orgId);
       } else {
+
+        // Get platform organization details
         getOrgAgent = await this.agentServiceRepository.getAgentDetails(platformAdminOrgDetails);
+
+        // Get platform organization data by orgId
         orgData = await this.agentServiceRepository.getOrgDetails(platformAdminOrgDetails);
       }
 
       agentSpinupDto.ledgerId = agentSpinupDto.ledgerId?.length ? agentSpinupDto.ledgerId : ledgerIdData.map(ledger => ledger?.id);
-      const ledgerDetails = await this.agentServiceRepository.getGenesisUrl(agentSpinupDto.ledgerId);
 
-      if (AgentSpinUpStatus.COMPLETED === getOrgAgent?.agentSpinUpStatus) {
-        throw new BadRequestException('Your wallet has already been created.');
-      }
+      // Get genesis URL and ledger details
+      const ledgerDetails = await this.agentServiceRepository.getGenesisUrl(agentSpinupDto.ledgerId);
 
       if (AgentSpinUpStatus.PROCESSED === getOrgAgent?.agentSpinUpStatus) {
-        throw new BadRequestException('Your wallet is already processing.');
+        throw new BadRequestException(
+          ResponseMessages.agent.error.walletAlreadyProcessing,
+          { cause: new Error(), description: ResponseMessages.errorMessages.badRequest }
+        );
       }
 
+      if (AgentSpinUpStatus.COMPLETED === getOrgAgent?.agentSpinUpStatus) {
```

Comment:
Can we use Switch case over here?

Reply:
Using a switch statement is preferable when dealing with multiple 'if-else' cases. If there is only a single 'if' condition, then using an 'if' statement is appropriate. Can I proceed with this approach ?